Src/Test.UnitTests.Sarif.PatternMatcher/SearchSkimmerTests.cs

```diff
@@ -409,12 +422,117 @@ public void SearchSkimmer_ShouldThrowWhenRuleDoesNotHaveAnyRegularExpression()
                 Logger = logger
             };
 
-            SearchSkimmer skimmer = CreateSkimmer(definition);
+            SearchSkimmer skimmer = CreateSkimmer(definition, fileSystem: mockFileSystem.Object);
             Exception exception = Record.Exception(() => skimmer.Analyze(context));
             exception.Should().NotBeNull();
             exception.GetType().Should().Be(typeof(InvalidOperationException));
         }
 
+        [Fact]
+        public void SearchSkimmer_ShouldNotEvaluateTooLargeFiles()
+        {
+            var random = new Random();
+            int randomMaxFileSize = random.Next(1, int.MaxValue - 1);
+            long randomFileSize = (long)random.Next(1, int.MaxValue - 1);
+
+            var testCases = new[]
+            {
+                new {
+                    fileSize = randomFileSize,
+                    maxFileSize = (int)uint.MinValue + 1,
+                },
+                new {
+                    fileSize = long.MaxValue,
+                    maxFileSize = int.MinValue,
+                },
+                new {
+                    fileSize = long.MaxValue,
+                    maxFileSize = (int)uint.MinValue + 1,
+                },
+                new {
+                    fileSize = long.MaxValue,
+                    maxFileSize = int.MaxValue,
+                },
+                new {
+                    fileSize = (long)ulong.MinValue,
+                    maxFileSize = (int)0,
+                },
+                new {
+                    fileSize = (long)ulong.MinValue,
+                    maxFileSize = randomMaxFileSize,
+                },
+                new {
+                    fileSize = (long)ulong.MinValue,
+                    maxFileSize = int.MaxValue,
+                },
+                new {
+                    fileSize = randomFileSize,
+                    maxFileSize = int.MaxValue,
+                },
+            };
+
+            foreach (var testCase in testCases)
+            {
+                var mockFileSystem = new Mock<IFileSystem>();
+
+                mockFileSystem.Setup(x => x.FileInfoLength(It.IsAny<string>())).Returns(testCase.fileSize);
+
+                MatchExpression expr = CreateGuidDetectingMatchExpression();
+                SearchDefinition definition = CreateDefaultSearchDefinition(expr);
+
+                string filePath = $"file:///c:/{definition.Name}.{definition.FileNameAllowRegex}";
+
+                var logger = new TestLogger();
+
+                var context = new AnalyzeContext
+                {
+                    TargetUri = new Uri(filePath),
+                    Logger = logger,
+                    MaxFileSizeInKilobytes = testCase.maxFileSize
+                };
+
+                SearchSkimmer skimmer = CreateSkimmer(definition, fileSystem: mockFileSystem.Object);
+                Exception exception = Record.Exception(() => skimmer.Analyze(context));
```

Comment:
>skimmer.Analyze [](http://example.com/codeflow?start=61&length=15)

how are we validating that we got in the code-path where the file was analyzed or not?